### PR TITLE
Improvement/Adds temporal_hidden_size_past and temporal_hidden_size_future hyperparams to TiDEModel

### PR DESCRIPTION
**Pull request by eschibli**
Checklist before merging this PR:
- [x] Mentioned all issues that this PR fixes or addresses.
- [x] Summarized the updates of this PR under **Summary**.
- [x] Added an entry under **Unreleased** in the [Changelog](../CHANGELOG.md).

Implements #2408

### Summary

Adds hyperparameters controlling hidden layer size for the past covariate and future covariate feature encoders in `TiDEModel`. Previously these were controlled by `hidden_size`. 

### Other Information
Defaults were set to 64. In the original google-research implementation the default `temporal_hidden_size_past` was set to 64, and there was no future covariate encoder. 

Alternatively, to preserve default behavior, defaults could be set to 128. 

### Other comments: 

Oops I see my rebase failed horribly - should I drop this PR and do it over from a fresh branch?

### CODE REVIEW

File: darts/models/forecasting/tide_model.py [hunk 1]
```diff
@@ -375,6 +383,8 @@ def __init__(
         hidden_size: int = 128,
         temporal_width_past: int = 4,
         temporal_width_future: int = 4,
+        temporal_hidden_size_past: int = 64,
```
Comment (dennisbader) [inline darts/models/forecasting/tide_model.py:386]: Same comments here as above

File: darts/models/forecasting/tide_model.py [hunk 2]
```diff
@@ -424,11 +434,15 @@ def __init__(
...
             The width of the layers in the residual blocks of the encoder and decoder.
         temporal_width_past
-            The width of the layers in the past covariate projection residual block. If `0`,
+            The width of the output layer in the past covariate projection residual block. If `0`,
             will bypass feature projection and use the raw feature data.
         temporal_width_future
-            The width of the layers in the future covariate projection residual block. If `0`,
+            The width of the output layer in the future covariate projection residual block. If `0`,
             will bypass feature projection and use the raw feature data.
+        temporal_hidden_size_past
```
Comment (dennisbader) [inline darts/models/forecasting/tide_model.py:442]: mention here that by default, it uses the same value as `temporal_width_*`, but that it is recommended setting a lower value.

**Review by dennisbader — CHANGES_REQUESTED** [review]
Hi @eschibli, and thanks for this PR.

In order to make it backwards compatible (e.g. loading a TiDEModel save from an older Darts version into the new one), we need to change the default behavior of the hyperparmeters.

File: darts/models/forecasting/tide_model.py [hunk 3]
```diff
@@ -79,6 +79,8 @@ def __init__(
         temporal_decoder_hidden: int,
         temporal_width_past: int,
         temporal_width_future: int,
+        temporal_hidden_size_past: int,
+        temporal_hidden_size_future: int,
```
[the remark this answers is not in the document]
Comment (eschibli) [reply darts/models/forecasting/tide_model.py]: Change made

File: darts/models/forecasting/tide_model.py [hunk 4]
```diff
@@ -148,6 +154,8 @@ def __init__(
         self.dropout = dropout
         self.temporal_width_past = temporal_width_past
         self.temporal_width_future = temporal_width_future
+        self.temporal_hidden_size_past = temporal_hidden_size_past
+        self.temporal_hidden_size_future = temporal_hidden_size_future
```
[the remark this answers is not in the document]
Comment (eschibli) [reply darts/models/forecasting/tide_model.py]: Change made

File: darts/models/forecasting/tide_model.py (hunk 1 above)
Comment (eschibli) [reply darts/models/forecasting/tide_model.py]: Change made

File: darts/models/forecasting/tide_model.py (hunk 2 above)
Comment (eschibli) [reply darts/models/forecasting/tide_model.py:442]: Change made

**Review by dennisbader — APPROVED** [review]
Thanks for the updates @eschibli, looks great now 🚀 

I made a last update to `TiDEModel._check_ckpt_parameters()`, so that loading weights from a model will be backwards compatible.

Everything's ready now, and I'll merge once all tests have passed 💯

**Pull request merged**